src/components/Files/ExcalidrawPage.tsx

```diff
@@ -0,0 +1,185 @@
+import { Excalidraw } from "@excalidraw/excalidraw";
+import { type ExcalidrawElement } from "@excalidraw/excalidraw/types/element/types";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { t } from "i18next";
+import { navigate } from "raviger";
+import { useEffect, useState } from "react";
+import { toast } from "sonner";
+
+import { debounce } from "@/lib/utils";
+
+import CareIcon from "@/CAREUI/icons/CareIcon";
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+
+import Loading from "@/components/Common/Loading";
+import { CreateFileResponse } from "@/components/Patient/models";
+
+import routes from "@/Utils/request/api";
+import mutate from "@/Utils/request/mutate";
+import query from "@/Utils/request/query";
+
+type Props = {
+  associatingId: string;
+  fileType: string;
+  drawingId?: string;
+};
+
+export default function ExcalidrawPage({
+  associatingId,
+  drawingId,
+  fileType,
+}: Props) {
+  const [elements, setElements] = useState<readonly ExcalidrawElement[] | null>(
+    drawingId ? null : [],
+  );
+  const [name, setName] = useState("");
+  const [id, setId] = useState(drawingId);
+  const [isDirty, setIsDirty] = useState(false);
+
+  const { data, isLoading } = useQuery({
+    queryKey: ["file", id],
+    queryFn: query(routes.retrieveUpload, {
+      pathParams: { id: id || "" },
+    }),
+    enabled: !!id,
+  });
+
+  useEffect(() => {
+    setIsDirty(!!elements?.length);
+  }, [elements?.length]);
+
+  useEffect(() => {
+    if (!data) {
+      return;
+    }
+    setName(data.name!);
+    const fetchData = async () => {
+      const response = await fetch(data.read_signed_url!);
+      const json = await response.json();
+      setElements(json.elements);
+    };
+    fetchData();
+  }, [data]);
+  const { mutateAsync: markUploadComplete, error: markUploadCompleteError } =
+    useMutation({
+      mutationFn: mutate(routes.markUploadCompleted, {
+        pathParams: { id: id || "" },
+      }),
+    });
+
+  const { mutateAsync: createUpload } = useMutation({
+    mutationFn: mutate(routes.createUpload),
+    onSuccess: (response: CreateFileResponse) => {
+      setId(response.id);
+    },
+  });
+
+  const handleSave = async () => {
+    if (!name.trim()) {
+      toast.error("Please enter a name for the drawing.");
+      return;
+    }
+
+    const obj = {
+      type: "excalidraw",
+      version: "2",
+      source: "https://care.ohc.network.com",
+      elements: elements,
+      appState: {},
+      files: {},
+    };
+
+    try {
+      const file = new File([JSON.stringify(obj)], `${name}.excalidraw`, {
+        type: "text/plain",
```

Comment:
ideally we should be using `application/vnd.excalidraw` or atleast `application/json`.